### PR TITLE
Added branch list fix: #459

### PR DESCRIPTION
**Pull request by codingtwinky**
Added branch list, this is same as #459.

* ~~Display text is "@<<current branch>>".  I'm not sure if this is a good idea but simply displaying branch name wasn't indicative enough.~~ having branch icon in the box should be indicative enough
* Branch list sorted

screen shot 2015-05-05 at 7 42 28 pm
screen shot 2015-05-05 at 7 41 56 pm
screen shot 2015-05-05 at 7 42 17 pm

### CODE REVIEW

File: components/branches/branches.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  return new BranchesViewModel(args.server, args.repoPath);
+});
+
+function BranchesViewModel(server, repoPath) {
+  var self = this;
+  this.repoPath = repoPath;
+  this.server = server;
+  this.branches = ko.observableArray();
+  this.fetchingProgressBar = components.create('progressBar', { predictionMemoryKey: 'fetching-' + this.repoPath, temporary: true });
+  this.fetchEnabled = true;
```
Comment (campersau) [inline components/branches/branches.js]: unused variable

File: components/branches/branches.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      var sorted = branches.sort(function(a, b) {
+        if (a.name < b.name)
+           return -1;
+        if (a.name > b.name)
+          return 1;
+        return 0;
+      });
+      self.branches(sorted);
+      self.current(undefined);
+      branches.map(function(branch) {
```
Comment (campersau) [inline components/branches/branches.js]: I would use `.forEach` for that. So no new array gets allocated.

**Comment by FredrikNoren** [comment]
This is really cool and much needed, but the row of stuff is getting a little too crazy (small displays will have a problem here).

How about we put all of this (submodules+branches) in a dialog instead, and just hide it in a [...] button? Like this:

screen shot 2015-08-27 at 9 05 17 pm

Both of them are kind of "things you need but not all the time" anyway, so I think it would make sense to put them in a place that you can get to but that's not on the "front page".

**Comment by codingtwinky** [comment]
Buttons now expand on hover...  I'm not a big fan of bundling buttons together... :P

screen shot 2015-08-29 at 5 06 22 pm

File: clicktests/test.noheader.js [hunk 3]
```diff
@@ -43,6 +43,10 @@ suite.test('Check for refresh button', function(done) {
   });
 });
 
+suite.test('Shutdown', function(done) {
```
Comment (campersau) [inline clicktests/test.noheader.js]: Isn't this the same as [below](https://github.com/FredrikNoren/ungit/pull/550/files#diff-adc25198ac2351b2aa9cc1351a87804bR51)?

**Comment by FredrikNoren** [comment]
That's really neat, I like that!

**Pull request merged**